File: reto8/reto8.py
def draw_race(indices, length):
    result = ""
    num_lanes = len(indices)

    for i in range(1, num_lanes + 1):
        spaces = " " * (num_lanes - i)
        lane = "~" * length

        if indices[i - 1] != 0:
            pos = indices[i - 1] if indices[i - 1] > 0 else length + indices[i - 1]
            lane = lane[:pos] + "r" + lane[pos + 1 :]

        result += f"{spaces}{lane} /{i}"
        if i < num_lanes:
            result += "\n"

    return result

File: reto8/reto8.ts
function drawRace(indices: number[], length: number): string {
  const numLanes = indices.length;
  let race = "";
  for (let i = 1; i <= numLanes; i++) {
    const spaces = " ".repeat(numLanes - i);
    const numLane = ` /${i}`;
    let lane = "~".repeat(length);
    if (indices[i - 1] !== 0) {
      const laneArray = lane.split("");
      laneArray.splice(indices[i - 1], 1, "r");
      lane = laneArray.join("");
    }
    race += spaces + lane + numLane;
    if (i !== numLanes) race += "\n";
  }
  return race;
}
